src/wireviz/wv_html.py

```diff
@@ -2,54 +2,88 @@
 # -*- coding: utf-8 -*-
 
 from pathlib import Path
-from typing import List, Union
+from typing import Dict, List, Union
 import re
 
 from wireviz import __version__, APP_NAME, APP_URL, wv_colors
 from wireviz.DataClasses import Metadata, Options
-from wireviz.wv_helper import flatten2d, open_file_read, open_file_write
+from wireviz.wv_helper import flatten2d, open_file_read, open_file_write, smart_file_resolve
+from wireviz.wv_gv_html import html_line_breaks
 
 def generate_html_output(filename: Union[str, Path], bom_list: List[List[str]], metadata: Metadata, options: Options):
+
+    # load HTML template
+    if 'name' in metadata.get('template',{}):
+        # if relative path to template was provided, check directory of YAML file first, fall back to built-in template directory
+        templatefile = smart_file_resolve(f'{metadata["template"]["name"]}.html', [Path(filename).parent, Path(__file__).parent / 'templates'])
+    else:
+        # fall back to built-in simple template if no template was provided
+        templatefile = Path(__file__).parent / 'templates/simple.html'
+
+    with open_file_read(templatefile) as file:
+        html = file.read()
+
+    # embed SVG diagram
+    with open_file_read(f'{filename}.svg') as file:
+        svgdata = file.read()
+        svgdata = re.sub(
+                  '^<[?]xml [^?>]*[?]>[^<]*<!DOCTYPE [^>]*>',
+                  '<!-- XML and DOCTYPE declarations from SVG file removed -->',
+                  svgdata, 1)
+
+    # generate BOM table
+    bom = flatten2d(bom_list)
+
+    # generate BOM header (may be at the top or bottom of the table)
+    bom_header_html = '  <tr>\n'
+    for item in bom[0]:
+        th_class = f'bom_col_{item.lower()}'
+        bom_header_html = f'{bom_header_html}    <th class="{th_class}">{item}</th>\n'
+    bom_header_html = f'{bom_header_html}  </tr>\n'
+
+    # generate BOM contents
+    bom_contents = []
+    for row in bom[1:]:
+        row_html = '  <tr>\n'
+        for i, item in enumerate(row):
+            td_class = f'bom_col_{bom[0][i].lower()}'
+            row_html = f'{row_html}    <td class="{td_class}">{item}</td>\n'
+        row_html = f'{row_html}  </tr>\n'
+        bom_contents.append(row_html)
+
+    bom_html = '<table>\n' + bom_header_html + ''.join(bom_contents) + '</table>\n'
+    bom_html_reversed = '<table>\n' + ''.join(list(reversed(bom_contents))) + bom_header_html + '</table>\n'
+
+    # prepare simple replacements
+    replacements = {
+        '<!-- %generator% -->': f'{APP_NAME} {__version__} - {APP_URL}',
+        '<!-- %fontname% -->': options.fontname,
+        '<!-- %bgcolor% -->': options.bgcolor,  # TODO: currently not working, requires translation from 2-char color code to HTML
```

Comment:
I will think about how to handle both 2-character color codes as well as HTML colors here. Any ideas are welcome.

Reply:
> I will think about how to handle both 2-character color codes as well as HTML colors here. Any ideas are welcome.

Why not keep the existing translation code as used before this PR (suggesting lowercase here, though):
```suggestion
        '<!-- %bgcolor% -->': wv_colors.translate_color(options.bgcolor, "hex"),
```
The HTML colors will be handled by a future PR to solve #135. In that future PR, I suggest we consider adding two new `ColorMode` entries `'html'` and `'HTML'` that returns a color name when an unknown name is specified, or the input matches the hex encoding of a known name, or returning the hex encoding otherwise.

Ideally, the lowercase/uppercase choice should follow the style of the template. Do we need both `<!-- %bgcolor% -->` and `<!-- %BGCOLOR% -->` for that, or do you see a more elegant solution?

Reply:
I don't want to over-complicate things so `<!-- %bgcolor% -->` and `hex` are enough.